ui/src/components/Model.js

```diff
@@ -40,35 +40,42 @@ const {
 } = base;
 
 const HorizontalTable = ({
-  fieldNames,
+  fieldNames = [],
   rawData,
-  extended,
+  extended: fieldData = [],
   customUnits = {},
   customValue = {},
-  data = (extended || [])
-    .slice()
-    .sort((a, b) => (fieldNames || []).indexOf(a) - (fieldNames || []).indexOf(b))
-    .reduce((acc, { field, type, displayName, unit }) => {
-      const fieldHelper = ({ field, type, displayName, unit }) =>
-        fieldNames.includes(field)
-          ? {
-              ...acc,
-              [field]: {
-                key: displayName,
-                value: apiDataProcessor({ data: get(rawData, field), type, unit }),
-              },
-            }
-          : acc;
-      return !Object.keys(customUnits).includes(field)
-        ? fieldHelper({ field, type, displayName, unit })
-        : fieldHelper({ field: field, type, displayName, unit: customUnits[field] || unit });
-    }, {}),
```

Comment:
This was so wild to me, this whole function definition in the default value declaration